lib/api.rb

```diff
@@ -285,6 +421,50 @@ def validation_error(msg)
     end
   end
 
+
+  # update a cipher via web vault
+  post "/ciphers/:uuid" do
+	response['access-control-allow-origin'] = '*'
+    d = device_from_bearer
+    if !d
+      return validation_error("invalid bearer")
+    end
+
+    c = nil
+    if params[:uuid].blank? ||
+    !(c = Cipher.find_by_user_uuid_and_uuid(d.user_uuid, params[:uuid]))
+      return validation_error("invalid cipher")
+    end
+
+    need_params(:type, :name) do |p|
+      return validation_error("#{p} cannot be blank")
+    end
+
+    begin
+      Bitwarden::CipherString.parse(params[:name])
+    rescue Bitwarden::InvalidCipherString
+      return validation_error("Invalid name")
+    end
+
+    if !params[:folderid].blank?
+      if !Folder.find_by_user_uuid_and_uuid(d.user_uuid, params[:folderid])
+        return validation_error("Invalid folder")
+      end
+    end
+
+    c.update_from_params(params)
+
+    Cipher.transaction do
+      if !c.save
+        return validation_error("error saving")
+      end
+
+      c.to_hash.merge({
+        "Edit" => true,
+      }).to_json
+    end
+  end
```

Comment:
This does the same thing as `put "/ciphers/:uuid"` just with a different HTTP verb, so both should just call a common function.